In [1]:
import pandas as pd
import numpy as np

In [3]:
df = pd.read_csv('netflix_titles.csv')

In [5]:
df.head(2)

show_id     type                 title         director  \
0      s1    Movie  Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show         Blood & Water              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   

                                         listed_in  \
0                                    Documentaries   
1  International TV Shows, TV Dramas, TV Mysteries   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...

In [6]:
df.tail(2)

show_id   type   title      director  \
8805   s8806  Movie    Zoom  Peter Hewitt   
8806   s8807  Movie  Zubaan   Mozez Singh   

                                                   cast        country  \
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

            date_added  release_year rating duration  \
8805  January 11, 2020          2006     PG   88 min   
8806     March 2, 2019          2015  TV-14  111 min   

                                           listed_in  \
8805              Children & Family Movies, Comedies   
8806  Dramas, International Movies, Music & Musicals   

                                            description  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...

In [7]:
df.shape

(8807, 12)

###  Data Processing

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


In [12]:
df.isnull().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

In [13]:
df.duplicated().sum()

0

#### Key Points 
* There are some feature that has null values:
    * Director = 2634
    * Cast = 825
    * Country = 831
    * Date_added = 10
    * Rating = 4
    * Duration = 3
* And there is no duplicate values present in the data set

#### Data Cleaning

In [17]:
percentage = round((df.isnull().sum()/df.shape[0])*100,2)

In [18]:
percentage

show_id          0.00
type             0.00
title            0.00
director        29.91
cast             9.37
country          9.44
date_added       0.11
release_year     0.00
rating           0.05
duration         0.03
listed_in        0.00
description      0.00
dtype: float64

In [19]:
df.dropna(subset=['duration','rating','date_added'], inplace=True)

In [20]:
df.isnull().sum()

show_id            0
type               0
title              0
director        2621
cast             825
country          829
date_added         0
release_year       0
rating             0
duration           0
listed_in          0
description        0
dtype: int64

In [22]:
df.shape

(8790, 12)

In [23]:
df.loc[:,'director'] = df.loc[:,'director'].fillna('Unavailable')
df.loc[:,'cast'] = df.loc[:,'cast'].fillna('Unavailable')
df.loc[:,'country'] = df.loc[:,'country'].fillna('Unavailable')

In [24]:
df.isnull().sum()

show_id         0
type            0
title           0
director        0
cast            0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
description     0
dtype: int64

#### Looks like we are done with the cleaning part

In [25]:
df['type'].unique()

array(['Movie', 'TV Show'], dtype=object)

In [27]:
df['country'].nunique()

749

In [30]:
df['rating'].unique()

array(['PG-13', 'TV-MA', 'PG', 'TV-14', 'TV-PG', 'TV-Y', 'TV-Y7', 'R',
       'TV-G', 'G', 'NC-17', 'NR', 'TV-Y7-FV', 'UR'], dtype=object)